internal/pkg/vm/internal/vmcontext/syscalls.go

```diff
@@ -0,0 +1,92 @@
+package vmcontext
+
+import (
+	ffi "github.com/filecoin-project/filecoin-ffi"
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/proofs/verification"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
+	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/internal/runtime"
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/crypto"
+	specsruntime "github.com/filecoin-project/specs-actors/actors/runtime"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/ipfs/go-cid"
+	"github.com/minio/blake2b-simd"
+)
+
+// Dragons: have the VM take a SysCalls object on construction (it will need a wrapper to charge gas)
+type syscalls struct {
+	gasTank *GasTracker
+}
+
+var _ specsruntime.Syscalls = (*syscalls)(nil)
+
+// VerifySignature implements Syscalls.
+func (sys syscalls) VerifySignature(signature crypto.Signature, signer address.Address, plaintext []byte) bool {
+	return types.IsValidSignature(plaintext, signer, signature)
+}
+
+// HashBlake2b implements Syscalls.
+func (sys syscalls) HashBlake2b(data []byte) [32]byte {
+	return blake2b.Sum256(data)
+}
+
+// ComputeUnsealedSectorCID implements Syscalls.
+func (sys syscalls) ComputeUnsealedSectorCID(sectorSize abi.SectorSize, pieces []abi.PieceInfo) (cid.Cid, error) {
+	panic("TODO")
+}
+
+// VerifySeal implements Syscalls.
+func (sys syscalls) VerifySeal(sectorSize abi.SectorSize, info abi.SealVerifyInfo) bool {
+	verifier := verification.NewFFIBackedProofVerifier()
+
+	commD := to32bytes(info.UnsealedCID.Bytes())
+	commR := to32bytes(info.OnChain.SealedCID.Bytes())
+
+	minerAddr, err := address.NewIDAddress(uint64(info.Miner))
+	if err != nil {
+		runtime.Abortf(exitcode.SysErrInternal, "failed to build ID address. %w", err)
+	}
+	miner := to32bytes(minerAddr.Bytes())
+
+	ticket := to32bytes([]byte(info.Randomness))
+	seed := to32bytes([]byte(info.InteractiveRandomness))
+	// Review: why the "ProofBytes" field? are we expecting other fields?
+	proof := []byte(info.OnChain.Proof.ProofBytes)
+
+	ok, err := verifier.VerifySeal(uint64(sectorSize), commR, commD, miner, ticket, seed, uint64(info.SectorID.Number), proof)
+	if err != nil {
+		runtime.Abortf(exitcode.SysErrInternal, "failed to validate PoRep: %w", err)
+	}
+
+	return ok
+}
+
+// VerifyPoSt implements Syscalls.
+func (sys syscalls) VerifyPoSt(sectorSize abi.SectorSize, info abi.PoStVerifyInfo) bool {
+	verifier := verification.NewFFIBackedProofVerifier()
+
+	sectorInfo := ffi.SortedPublicSectorInfo{}
+
+	randomness := to32bytes(info.Randomness)
+	// Dragons: where is this coming from?
+	challengeCount := uint64(21)
```

Comment:
Apparently the go-sector-builder has some of these answers. Maybe we should be invoking that instead.